### PR TITLE
HPCC 30381 Parquet plugin function names and member variables should be more consistent.

### PR DESCRIPTION
**Pull request by jackdelv**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: plugins/parquet/parquetembed.cpp [hunk 1]
```diff
@@ -497,169 +328,282 @@ arrow::Status ParquetHelper::processReadFile()
...
+    nextTable = &parquetTable;
+    totalRowsProcessed++;
+    return rowsProcessed++;
+}
+
+/**
+ * @brief Contructs a ParquetWriter for the target location and checks for existing data.
+ *
+ * @param option The read or write option as well as information about partitioning.
+ * @param _destination The destination to write a parquet file.
```
Comment (dcamper) [inline plugins/parquet/parquetembed.cpp:412]: That is a full path, correct?

File: plugins/parquet/parquetembed.cpp [hunk 2]
```diff
@@ -497,169 +328,282 @@ arrow::Status ParquetHelper::processReadFile()
...
+ * @param _rowSize The max row group size when creating RecordBatches for output.
```
Comment (dcamper) [inline plugins/parquet/parquetembed.cpp:413]: What is the scale?  Bytes?

File: plugins/parquet/parquetembed.cpp [hunk 3]
```diff
@@ -497,169 +328,282 @@ arrow::Status ParquetHelper::processReadFile()
...
-    rbatchItr++;
-    std::vector<std::shared_ptr<arrow::RecordBatch>> to_table = {batch};
-    return std::move(arrow::Table::FromRecordBatches(std::move(to_table)));
+    if (++currentRow == rowSize)
+        currentRow = 0;
 }
 
-std::unordered_map<std::string, std::shared_ptr<arrow::Array>> &ParquetHelper::next()
+/**
+ * @brief convert a vector of rapidjson::Documents containing single rows to an arrow::RecordBatch
```
Comment (dcamper) [inline plugins/parquet/parquetembed.cpp:571]: Minor: Capital 'convert'

File: plugins/parquet/parquetembed.cpp [hunk 4]
```diff
@@ -1744,68 +1700,80 @@ void ParquetRecordBinder::processDecimal(const void *value, unsigned digits, uns
...
-    bool overwrite = false;     // If true overwrite file with no error. The default is false and will throw an error if the file already exists.
-    arrow::Compression::type compressionOption = arrow::Compression::UNCOMPRESSED;
+    const char *option = "";            // Read(read), Read Parition(readpartition), Write(write), Write Partition(writepartition)
+    const char *location = "";          // file name and location of where to write parquet file
+    const char *destination = "";       // file name and location of where to read parquet file from
+    const char *partitionFields = "";   // Semicolon delimited values containing fields to partition files on
+    __int64 rowSize = 40000;            // Size of the row groups when writing to parquet files
+    __int64 batchSize = 40000;          // Size of the batches when converting parquet columns to rows
+    bool overwrite = false;             // If true overwrite file with no error. The default is false and will throw an error if the file already exists.
+    arrow::Compression::type compressionOption = arrow::Compression::UNCOMPRESSED; // Compression option that supports all arrow compression types.
```
Comment (dcamper) [inline plugins/parquet/parquetembed.cpp:1776]: The comment is a bit confusing.  I think it is referring to 'what is compressionOption' but it somehow makes me think it is describing arrow::Compression::UNCOMPRESSED

**Review by dcamper — CHANGES_REQUESTED** [review]
A few minor changes, all in comments.  Looks good!

File: plugins/parquet/parquetembed.cpp (hunk 4 above)
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp]: Reworded description of compressionOption.

File: plugins/parquet/parquetembed.cpp (hunk 3 above)
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp]: Fixed

File: plugins/parquet/parquetembed.cpp [hunk 5]
```diff
@@ -497,169 +328,282 @@ arrow::Status ParquetHelper::processReadFile()
...
+        splitTable(table);
     }
-    else if (partOption[0] == 'R' || partOption[0] == 'r')
+    nextTable = &parquetTable;
+    totalRowsProcessed++;
+    return rowsProcessed++;
+}
+
+/**
+ * @brief Contructs a ParquetWriter for the target location and checks for existing data.
```
[the remark this answers is not in the document]
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp]: Fixed

File: plugins/parquet/parquetembed.cpp [hunk 6]
```diff
@@ -97,38 +97,20 @@ extern void fail(const char *message)
...
- * @param location The location to read a parquet file.
- *
- * @param destination The destination to write a parquet file.
- *
- * @param rowsize The max row group size when reading parquet files.
+ * @brief Contructs a ParquetReader for a specific file location.
  *
+ * @param option The read or write option as well as information about partitioning.
+ * @param _location The location to read a parquet file.
  * @param _batchSize The size of the batches when converting parquet columns to rows.
```
[the remark this answers is not in the document]
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp]: batchSize is the number of rows in the RecordBatch. Changed comment to be more descriptive.

File: plugins/parquet/parquetembed.cpp [hunk 7]
```diff
@@ -97,38 +97,20 @@ extern void fail(const char *message)
...
- *
- * @param location The location to read a parquet file.
- *
- * @param destination The destination to write a parquet file.
- *
- * @param rowsize The max row group size when reading parquet files.
+ * @brief Contructs a ParquetReader for a specific file location.
  *
+ * @param option The read or write option as well as information about partitioning.
+ * @param _location The location to read a parquet file.
```
[the remark this answers is not in the document]
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp]: Capitalized Parquet and reworded to clarify it is a full path to the target.

File: plugins/parquet/parquetembed.cpp (hunk 2 above)
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp]: Reworded to clarify that rowSize is the maximum number of rows in a RecordBatch.

Comment (dcamper) [reply plugins/parquet/parquetembed.cpp]: May I suggest maxRowCountInBatch or something along those lines?

Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp]: Changed to maxRowCountInBatch. That is much more readable and causes less confusion.

**Review by dcamper — CHANGES_REQUESTED** [review]
A few very minor changes.  Looking really good!

File: plugins/parquet/parquetembed.cpp [hunk 8]
```diff
@@ -406,20 +406,20 @@ __int64 ParquetReader::next(TableColumns *&nextTable)
...
 
 /**
- * @brief Contructs a ParquetWriter for the target location and checks for existing data.
+ * @brief Constructs a ParquetWriter for the target location and checks for existing data.
  *
  * @param option The read or write option as well as information about partitioning.
- * @param _destination The destination to write a parquet file.
- * @param _rowSize The max row group size when creating RecordBatches for output.
+ * @param _destination The full path for which to write a Parquet file or partitioned dataset.
+ * @param _maxRowCountInBatch The max number of rows when creating RecordBatches for output.
```
[the remark this answers is not in the document]
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp:413]: Changed batchSize to maxRowCountInTable.

File: plugins/parquet/parquetembed.cpp [hunk 9]
```diff
@@ -1767,13 +1766,13 @@ ParquetEmbedFunctionContext::ParquetEmbedFunctionContext(const IContextLogger &_
 {
     // Option Variables
     const char *option = "";            // Read(read), Read Parition(readpartition), Write(write), Write Partition(writepartition)
-    const char *location = "";          // file name and location of where to write parquet file
-    const char *destination = "";       // file name and location of where to read parquet file from
+    const char *location = "";          // file name and location of where to write Parquet file
+    const char *destination = "";       // file name and location of where to read Parquet file from
     const char *partitionFields = "";   // Semicolon delimited values containing fields to partition files on
```
[the remark this answers is not in the document]
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp:1771]: Fixed.

**Review by dcamper — APPROVED** [review]
Looks good!  Please squash.

**Comment by jackdelv** [comment]
@dcamper Squashed.

**Comment by jackdelv** [comment]
@ghalliday This is ready to merge.

**Comment by ghalliday** [comment]
@jackdelv I should have checked.  For the future, the format of the commit should be "HPCC-NNNN" rather than "HPCC NNNN".

**Pull request merged**